src/network.js

```diff
@@ -536,23 +540,37 @@ Network.prototype = {
     if (!this.optimized)
       this.optimize();
 
-    var hardcode = "var inputs = " + this.optimized.data.inputs.length + ";\n";
-    hardcode += "var outputs = " + this.optimized.data.outputs.length + ";\n";
-    hardcode += "var F =  new Float64Array([" + this.optimized.memory.toString() + "]);\n";
-    hardcode += "var activate = " + this.optimized.activate.toString() + ";\n";
-    hardcode += "var propagate = " + this.optimized.propagate.toString() + ";\n";
-    hardcode +=
-        "onmessage = function(e) {\n" +
-          "if (e.data.action == 'startTraining') {\n" +
-            "train(" + JSON.stringify(set) + "," + JSON.stringify(workerOptions) + ");\n" +
-          "}\n" +
-        "}";
-
-    var workerSourceCode = workerFunction + '\n' + hardcode;
-    var blob = new Blob([workerSourceCode]);
-    var blobURL = window.URL.createObjectURL(blob);
-
-    return new Worker(blobURL);
+    var hardcode =
+      "var F = new Float64Array([" + this.optimized.memory.join(',') + "]);\n" +
```

Comment:
Can you please keep the original order of lines? It helps to understand what was really changed. Different commit with this change will be greatly appreciated, still each of commits would preferred to be as minimal-impact as possible.